src/App.js

```diff
@@ -1,25 +1,79 @@
-import logo from './logo.svg';
-import './App.css';
+import { Fragment, useEffect, useContext } from 'react'
+import { Redirect, Route, Switch } from 'react-router-dom'
+import { useHistory } from 'react-router';
+import MainHeader from './components/Layout/MainHeader';
+import Homepage from './Pages/Homepage';
+import Dashboard from './Pages/Dashboard';
+import Login from './Pages/Login';
+import NotFound from './Pages/NotFound';
+import Profile from './Pages/Profile';
+import Todos from './Pages/Todos';
+import TodoDetail from './Pages/TodoDetail';
+import AddTodo from './Pages/AddTodo';
+import { AuthContext } from './store/auth-context';
+import Register from './Pages/Register';
+import { TodoContext } from "./store/todo-context";
+
+
+const App = () => {
+  const history = useHistory();
+  const authCtx = useContext(AuthContext);
+  const todoCtx = useContext(TodoContext);
+  const todos = todoCtx.todos;
+  const user = authCtx.user;
+  const userId = user?.id;
+
+  useEffect(() => {
+    if (userId) {
+      todoCtx.fetchTodo();
+    }
+  }, [userId])
```

Comment:
Taky - fetchoval bych data až v té komponentě, ve které je potřebuju.

To je to, kde jsi psal, že se ti posílá fetch 2x, že? Musel bych si to rozjet u sebe, abych vyzkoušel, co se tam děje. Ale podezřívám, že by ta hodnota `user` v Contextu mohla být vytvářena při každém renderu, což by donutilo `useEffect` znova spustit.